src/wp-includes/link-template.php

```diff
@@ -2949,14 +2949,20 @@ function get_the_posts_pagination( $args = array() ) {
 			$args['aria_label'] = $args['screen_reader_text'];
 		}
 
+		if ( ! empty( get_queried_object()->labels->name ) ) {
+			$post_type_name = get_queried_object()->labels->name;
+		} else {
+			$post_type_name = 'Posts';
+		}
+
 		$args = wp_parse_args(
 			$args,
 			array(
 				'mid_size'           => 1,
 				'prev_text'          => _x( 'Previous', 'previous set of posts' ),
 				'next_text'          => _x( 'Next', 'next set of posts' ),
-				'screen_reader_text' => __( 'Posts navigation' ),
-				'aria_label'         => __( 'Posts' ),
+				'screen_reader_text' => sprintf( __( '%s navigation' ), $post_type_name ),
+				'aria_label'         => sprintf( __( '%s pagination' ), $post_type_name ),
```

Comment:
```suggestion
				/* translators: %s: Post type name. */
				'screen_reader_text' => sprintf( __( '%s navigation' ), $post_type_name ),
				/* translators: %s: Post type name. */
				'aria_label'         => sprintf( __( '%s pagination' ), $post_type_name ),
```

Add [translator comments](https://github.com/WordPress/wordpress-develop/blob/d23f60f5b77de0d116b2abed3094989ae15874c3/src/wp-includes/post.php#L2101-L2102) to explain the `%s` placeholder.